### DISCUSSION TITLE
Format all numbers with comma thousands separator

### DISCUSSION THREAD

**Discussion by maia-sh** [quarto-dev/quarto-cli#6530] [Q&A] [answered]
### Description

I would like to to format all numbers in my Quarto document using a comma thousands separator. For example, inline code that evaluates to `1500` should print as `1,500`.

Currently, I need to wrap each number in a function to apply the formatting, e.g. `r scales::comma(1500)`.

Ideally, I would like a document-level YAML option such as `number-format` to set all numbers.
~~I found the similar option for dates: `date-format` (see https://quarto.org/docs/reference/dates.html)~~ (CORRECTION: This specifies the format of the `date` YAML field and NOT date throughout the document.)

Is there such an option that I'm not finding in the documentation?
Otherwise, happy to hear others' workarounds!

**Comment by rgaiacs** [comment]
Dear @maia-sh,

Because you are using `knitr`, you can use 

```
knit_hooks$set(inline = function(x) {
  prettyNum(x, big.mark=" ")
})
```

at the begin of your documents as mentioned at https://stackoverflow.com/a/18967590/1802726.

**Reply by maia-sh** [reply, marked as answer]
Many thanks, @rgaiacs! I wasn't familiar with knitr hooks. This solution works, however, sweeps up non-numeric variables, i.e, dates.

Based on your suggestion as well as a [stackoverflow discussion on formatting all dates](https://stackoverflow.com/questions/70793780/set-date-format-for-knitr-document), I added to the top of my Quarto document:



```r
# Format integers
knit_print.integer = function(x, ...) {
  prettyNum(x, big.mark=",")
}

registerS3method(
  "knit_print", "integer", knit_print.integer,
  envir = asNamespace("knitr")
)

# Format dates
knit_print.Date = function(x, ...) {
  format(x, "%d %B %Y")
}

registerS3method(
  "knit_print", "Date", knit_print.Date,
  envir = asNamespace("knitr")
)
```

This prints `1,500` and `16 August 2023`.

**Comment by mcanouil** [comment]
Indeed, this is something do be done within the language you are using or write a Lua filter that parses strings to detect number and rewrite/format them.
